python/ray/dag/dag_node_operation.py

```diff
@@ -723,34 +614,18 @@ def _generate_overlapped_execution_schedule(
         "ray.actor.ActorHandle", List[_DAGOperationGraphNode]
     ] = copy.deepcopy(actor_to_execution_schedule)
     for overlapped_schedule in actor_to_overlapped_schedule.values():
-        for i in range(len(overlapped_schedule)):
+        for i in range(1, len(overlapped_schedule)):
```

Comment:
Since we don't have `READ`, `COMPUTE`, and `WRITE` nodes anymore, the condition for overlapping NCCL read and computation becomes simpler: if the node previous to a NCCL read node is a compute node, swap them. Since the first node (at index 0) does not have a previous node, we start at one.